tools/llvm-kompile-codegen/main.cpp

```diff
@@ -24,49 +31,133 @@
 #include <unordered_map>
 #include <utility>
 
+using namespace llvm;
 using namespace kllvm;
 using namespace kllvm::parser;
 
-std::string getFilename(
-    std::map<std::string, std::string> index, char **argv,
+namespace fs = std::filesystem;
+
+cl::OptionCategory CodegenCat("llvm-kompile-codegen options");
+
+cl::opt<std::string> Definition(
+    cl::Positional, cl::desc("<definition.kore>"), cl::Required,
+    cl::cat(CodegenCat));
+
+cl::opt<std::string> DecisionTree(
+    cl::Positional, cl::desc("<dt.yaml>"), cl::Required, cl::cat(CodegenCat));
+
+cl::opt<std::string> Directory(
+    cl::Positional, cl::desc("<dir>"), cl::Required, cl::cat(CodegenCat));
+
+cl::opt<int, true> Debug(
```

Comment:
let's go ahead and make this a flag rather than a positional argument since we're refactoring this.